app/src/main/java/de/tum/in/tumcampusapp/activities/GradesActivity.java

```diff
@@ -55,110 +66,61 @@ public GradesActivity() {
         super(TUMOnlineConst.Companion.getEXAMS(), R.layout.activity_grades);
     }
 
-    /**
-     * Builds HTML string showing a column chart
-     *
-     * @param filteredExamList List of exams
-     * @return content string
-     */
-    String buildColumnChartContentString(List<Exam> filteredExamList) {
-        Map<String, Integer> gradeDistribution = calculateGradeDistribution(filteredExamList);
-
-        StringBuilder datas = new StringBuilder(1024);
-        // Build data string
-        for (int i = 0; i < GRADES.length; i++) {
-            datas.append("['")
-                 .append(GRADES[i])
-                 .append("', ")
-                 .append(gradeDistribution.get(GRADES[i]))
-                 .append(']');
-
-            if (i != GRADES.length - 1) {
-                datas.append(',');
-            }
+    private int[] getGradeColors(){
+        return new int[]{
+                R.color.grade_1_0, R.color.grade_1_3, R.color.grade_1_3, R.color.grade_1_7,
```

Comment:
I don't think it's self explanatory why there is a duplicated `R.color.grade_1_3` in here
I'd define a separate color `R.color.grade_1_4` with the same value.

Also: this method can be a static constant